static/js/components/dashboard/courses/GradeDetailPopup.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import Dialog from 'material-ui/Dialog';
+import React from 'react';
+import R from 'ramda';
+import Icon from 'react-mdl/lib/Icon';
+
+import type {
+  Course,
+  CourseRun,
+} from '../../../flow/programTypes';
+import { formatGrade } from '../util';
+import {
+  STATUS_NOT_PASSED,
+  STATUS_PASSED,
+} from '../../../constants';
+
+const runGrade = (courseRun: CourseRun, isBestGrade) => {
+  if (courseRun.final_grade) {
+    if (isBestGrade) {
+      return <div className="best-grade">
+        { formatGrade(courseRun.final_grade) }
+        <Icon name="check" />
+      </div>;
+    }
+    return formatGrade(courseRun.final_grade);
+  }
+  return "";
+};
+
+const runStatus = courseRun => {
+  if (courseRun.status === STATUS_NOT_PASSED) {
+    return <div className="status failed">
+      Not passed
+    </div>;
+  }
+  if (courseRun.status === STATUS_PASSED || courseRun.passed === true) {
```

Comment:
I don't think `passed` is a property of `courseRun`. Also can you add a flow annotation for this argument?